srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+        sequence<octet, MAX_BOUNDED_SEQ_SIZE> bin_data;
+    };
+
+However, in order to use a given type, the struct has been modified as follows:
+
+.. code-block:: c
+
+    struct TestData_t {
+        octet key[KEY_SIZE];
+        long entity_id;
+        unsigned long seq_num;
+        long timestamp_sec;
+        unsigned long timestamp_usec;
+        long latency_ping;
+        sequence<octet, MAX_BOUNDED_SEQ_SIZE> bin_data;
+    #ifdef RTI_CUSTOM_TYPE
+        RTI_CUSTOM_TYPE custom_type;
+        // custom_type_size is the serialize size of the field custom_type
+        // in the the sample that we send on the wire
+        long custom_type_size;
+    #endif
+    };
+
+Notice that a `RTI_CUSTOM_TYPE` member has been added to the struct (conditionally).
+This placeholder is replaced at generation time with the type specified by the
+user as the custom type.
+
+Therefore if the customer provides an `.idl` file with the following structure:
+
+.. code-block:: c
+
+    struct Test {
+        sequence<octet, MAX_SIZE> test_seq;
+        long test_long; //@key
+    };
+
+The type Name needed for `RTI_CUSTOM_TYPE` will be `Test` (independently
+on the name of the file where the struct is, or if it is preceeded or followed by
+other structures).
+
+Steps to use Custom Types
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The steps needed to use a custom type in *RTI Perftest* are:
+
+Copy the IDL files into `~/rtiperftest/srcIdl/customTypes/`
+-----------------------------------------------------------
+
+There are some restrictions for this step:
+
+- You can include multiple `.idl` files, but all of them should be at the same
+  level (no folders are allowed inside the `customTypes` folder.
+- The `idl` files cannot be named `custom.idl`. A file with this name will be generated
+  automatically by the `build` script, so this file would be overwritten.
+- The main type you will link should be declared as ``struct <NameOfTheType> {``.
+  The `struct` and `{` words should be spaced as shows. This way the build scripts
+  will be able to find the type accross a list of `.idl`.
+
+Implement the API custom type functions of `customType.cxx`
+-----------------------------------------------------------
+
+You should find a file in `${PERFTEST_HOME}/srcCpp` named `customType.cxx`. This
+file contains several functions that must be implemented in order to correctly
+initialize, finalize and populate the sample.
+
+**Note:** *RTI Perftest* will not initialize by default sequences, optional members
+or non-primitive structures. This means that those fields will need to be
+initialized by the user in the `set_custom_type_data`, `set_custom_type_dynamic_data`
+and `set_custom_type_data_flatdata`.
+
+**Note:** If using *FlatData* see the dedicated section below.
+
+Run the `build` script adding ``--customType <NameOfTheType``
+-------------------------------------------------------------
+
+Run the ``build.sh`` or ``build.bat`` script using the ``--customType`` option.
```

Comment:
using the ``--customType <NameOfTheType>`` option.

Reply:
The title of this section is "Run the `build` script adding ``--customType <NameOfTheType``" and then the first line in this section is "Run the ``build.sh`` or ``build.bat`` script using the ``--customType`` option."  Seems a bit redundant.

Recommend

Building Perftest with custom types
-----------------------------------------------------------------------------------

Run the ``build.sh`` or ``build.bat`` script with the additional ``--customType <NameOfTheType>`` option.  For testing FlatData and/or ZeroCopy, please see the section **Custom Types + `FlatData`**.